src/Process.js

```diff
@@ -57,8 +57,8 @@ Process.prototype.ready = T.func([T.Number], T.Object, "process.ready").of(funct
     const deferred = Q.defer();
 
     this.instance.onReady(deferred.resolve);
-    this.instance.run();
-    setTimeout(deferred.reject, timeout, new Error("Timeout exceeded."));
+    this.instance.onTimeout(deferred);
```

Comment:
Hab deine Vorschläge aufgegriffen und ein bisschen verändert. M.E. ist das so konsistent, damit ist das single responsibility prinzip besser gedeckt und es wird konsistent der eventemitter verwendet.

onTimeout setzt damit nur ein callback das die deferred rejected. In der run funktion wird dann der timeout gesetzt, was eigentlich so sein sollte denke ich, da es erst sinn macht den Timeout zu setzen, sobald der Prozess überhaupt erst gestartet wurde.